arbor/include/arbor/util/any_ptr.hpp

```diff
@@ -62,7 +60,7 @@ struct ARB_SYMBOL_VISIBLE any_ptr {
     }
 
     template <typename T, typename = std::enable_if_t<std::is_pointer<T>::value>>
-    T as() const noexcept {
+    constexpr T as() const noexcept {
         if (std::is_same<T, void*>::value) {
```

Comment:
```suggestion
        if constexpr (std::is_same_v<T, void*>) {
```

Reply:
note: `std::type_info::operator==` becomes `constexpr` in c++23

Reply:
note2: since you are adding `constexpr` to some member functions, consider making all suitable member functions `constexpr`.